### ISSUE TITLE
Removing the bottom margin of the last child within a panel body

### ISSUE THREAD

**Issue by monoblaine** [twbs/bootstrap#16765]
Many (maybe all) elements (`p`, `ul`, `ol`, `dl`, `blockquote`, etc.) have some bottom margin, which is totally fine and reasonable most of the time. But when they are placed within the body (`panel-body`) of a `panel` component, I think it would be better if we remove the bottom margin of the last child, because the `panel-body` has already paddings on all four sides.

~~~less
.panel-body {
  & > :last-child {
    margin-bottom: 0;
  }
}
~~~

**This is how it currently looks like:**
https://jsbin.com/tovinikino/edit?html,css,output

**This is how it looks like when the bottom margin of the last child is removed:**
https://jsbin.com/juwepitafe/1/edit?html,css,output

**Comment by cvrebert** [comment]
Approximate duplicate of #10825. Please search more in the future.

**Comment by monoblaine** [comment]
I did search through the issues, but not in the pull requests. Sorry about that.

@mdo appears to have closed that issue (#10825) with the following explanation:

> Generally I'm against trying to remove anything from any last child within a parent component. I understand the appeal, but all I can see is folks running issues because their own CSS isn't enough to counteract the `:last-child` specificity.

However, even the web site of Bootstrap itself is full of such codes[*](https://github.com/twbs/bootstrap/blob/master/docs/assets/css/src/docs.css#L972):

~~~css
.bs-example > p:last-child,
.bs-example > ul:last-child,
.bs-example > ol:last-child,
.bs-example > blockquote:last-child,
.bs-example > .form-control:last-child,
.bs-example > .table:last-child,
.bs-example > .navbar:last-child,
.bs-example > .jumbotron:last-child,
.bs-example > .alert:last-child,
.bs-example > .panel:last-child,
.bs-example > .list-group:last-child,
.bs-example > .well:last-child,
.bs-example > .progress:last-child,
.bs-example > .table-responsive:last-child > .table {
  margin-bottom: 0;
}
~~~

I don't really understand the hesitation, and I doubt the idea that people will run into problems because they cannot see extra margins in their panels.
